src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -69,6 +69,14 @@ public ObjectResult(object value)
         public virtual IOutputFormatter SelectFormatter(OutputFormatterContext formatterContext,
                                                        IEnumerable<IOutputFormatter> formatters)
         {
+            if (ContentTypes.Count == 1)
+            {
+                // There is only one content type specified so we can skip looking at the accept headers.
+                return SelectFormatterUsingAnyAcceptableContentType(formatterContext,
+                                                                    formatters,
+                                                                    ContentTypes);
+            }
+
             var incomingAcceptHeaderMediaTypes = formatterContext.ActionContext.HttpContext.Request.GetTypedHeaders().Accept ??
```

Comment:
Yes - It's unrelated, but not worth a separate PR. It was super unclear what happens when there is a single content, and lots of extra code was being executed. Since this is a scenario in this PR, I moved it up. Not extra tests needed